tool/tctl/common/auth_command_test.go

```diff
@@ -46,35 +50,99 @@ func TestAuthSignKubeconfig(t *testing.T) {
 			TLS: []byte("TLS cert"),
 		},
 		cas: []services.CertAuthority{ca},
+		proxies: []services.Server{
+			&services.ServerV2{
+				Kind:    services.KindNode,
+				Version: services.V2,
+				Metadata: services.Metadata{
+					Name: "proxy",
+				},
+				Spec: services.ServerSpecV2{
+					PublicAddr: "proxy-from-api.example.com:3080",
+				},
+			},
+		},
 	}
-	ac := &AuthCommand{
-		output:       filepath.Join(tmpDir, "kubeconfig"),
-		outputFormat: identityfile.FormatKubernetes,
-		proxyAddr:    "proxy.example.com",
-	}
-
-	// Generate kubeconfig.
-	if err = ac.generateUserKeys(client); err != nil {
-		t.Fatalf("generating kubeconfig: %v", err)
+	tests := []struct {
+		desc     string
+		ac       AuthCommand
+		wantAddr string
+	}{
+		{
+			desc: "--proxy specified",
+			ac: AuthCommand{
+				output:       filepath.Join(tmpDir, "kubeconfig"),
+				outputFormat: identityfile.FormatKubernetes,
+				proxyAddr:    "proxy-from-flag.example.com",
+			},
+			wantAddr: "proxy-from-flag.example.com",
+		},
+		{
+			desc: "k8s proxy running locally with public_addr",
+			ac: AuthCommand{
+				output:       filepath.Join(tmpDir, "kubeconfig"),
+				outputFormat: identityfile.FormatKubernetes,
+				config: &service.Config{Proxy: service.ProxyConfig{Kube: service.KubeProxyConfig{
+					Enabled:     true,
+					PublicAddrs: []utils.NetAddr{{Addr: "proxy-from-config.example.com:3026"}},
+				}}},
+			},
+			wantAddr: "proxy-from-config.example.com:3026",
+		},
+		{
+			desc: "k8s proxy running locally without public_addr",
+			ac: AuthCommand{
+				output:       filepath.Join(tmpDir, "kubeconfig"),
+				outputFormat: identityfile.FormatKubernetes,
+				config: &service.Config{Proxy: service.ProxyConfig{
+					Kube: service.KubeProxyConfig{
+						Enabled: true,
+					},
+					PublicAddrs: []utils.NetAddr{{Addr: "proxy-from-config.example.com:3080"}},
```

Comment:
Is there a situation where we shouldn't be appending `https://` to the Kubernetes proxy address? I appreciate that this is just a unit test, but it might be more representative of reality if the test addresses also had the prefix.

Reply:
good point, changed the logic to always prepend `https://`